service/service.go

```diff
@@ -612,15 +627,38 @@ func (s *service) loadIsilonConfigs(ctx context.Context, configFile string) erro
 	return nil
 }
 
-func (s *service) reconcileNodeAzLabels(interval time.Duration) error {
-	_, log := GetLogger(context.Background())
+// getReconcileInterval returns the access zone reconcile interval
+func (s *service) getReconcileInterval() time.Duration {
+	syncMutex.Lock()
+	defer syncMutex.Unlock()
+	return s.azReconcileInterval
+}
+
+// getUpdateIntervalChannel returns the updated access zone reconcile interval
+func (s *service) getUpdateIntervalChannel() <-chan time.Duration {
+	return s.updateAZReconcileIntervalCh
+}
+
+// reconcileNodeAzLabels reconciles the node access zone labels
+func (r *reconciler) reconcileNodeAzLabels(ctx context.Context) error {
+	_, log := GetLogger(ctx)
+
 	go func() {
+		ticker := time.NewTicker(r.service.getReconcileInterval())
+		defer ticker.Stop()
+
 		for {
-			err := s.ReconcileNodeAzLabels(context.Background())
-			if err != nil {
-				log.Errorf("Node label reconciliation failed: %v", err)
+			select {
+			case <-ticker.C:
+				err := r.service.ReconcileNodeAzLabels(ctx)
+				if err != nil {
+					log.Errorf("node label reconciliation failed: %v", err)
+				}
+			case newInterval := <-r.service.getUpdateIntervalChannel():
+				ticker.Stop()
+				ticker = time.NewTicker(r.service.getReconcileInterval())
```

Comment:
Was there no API to reset with a new interval?

Reply:
No, once the ticker is created, it has to be stopped in order to use the new interval.

Reply:
What about Reset()? https://cs.opensource.google/go/go/+/refs/tags/go1.24.6:src/time/tick.go;l=65